m4/hts_hide_dynamic_syms.m4

```diff
@@ -0,0 +1,65 @@
+dnl @synopsis HTS_HIDE_DYNAMIC_SYMBOLS
+dnl
+dnl Turn on compiler options that prevent unwanted symbols from being exported
+dnl by shared libraries.
+dnl
+dnl @author Rob Davies <rmd@sanger.ac.uk>
+dnl @license MIT/Expat
+dnl
+dnl Copyright (C) 2018 Genome Research Ltd.
```

Comment:
Wrong year. :-)

(Or have we really had a branch for a year awaiting a PR to be made from it?)